extensions/binding/org.eclipse.smarthome.binding.weatherunderground/src/main/java/org/eclipse/smarthome/binding/weatherunderground/handler/WeatherUndergroundHandler.java

```diff
@@ -298,6 +298,9 @@ private WeatherUndergroundJsonData getWeatherData(Set<String> features, String l
             } else if (result.getResponse() == null) {
                 errorDetail = "missing response sub-object";
             } else if (result.getResponse().getErrorDescription() != null) {
+                if ("keynotfound".equals(result.getResponse().getErrorType())) {
```

Comment:
It is to have a clear message about wrong API key displayed in Paper UI (and something that can easily be translated). Consider that as an additional improvement of the binding.